x11-dl/src/link.rs

```diff
@@ -55,12 +55,11 @@ macro_rules! x11_link {
         unsafe {
           let libdir = $crate::link::config::libdir::$pkg_name;
           let lib = try!($crate::link::DynamicLibrary::open_multi(libdir, &[$($lib_name),*]));
-          let mut this: ::std::mem::ManuallyDrop<$struct_name>
-              = ::std::mem::uninitialized();
-          let this_ptr = &mut this as *mut _ as *mut $struct_name;
+          let mut this = ::maybe_uninit::MaybeUninit::<$struct_name>::uninit();
+          let this_ptr = this.as_mut_ptr();
           ::std::ptr::write(&mut (*this_ptr).lib, lib);
           try!(Self::init(this_ptr));
-          Ok(::std::mem::ManuallyDrop::into_inner(this))
+          Ok(this.assume_init())
```

Comment:
So, it is guaranteed then that `Self::init` *will* put non-NULL values into all these function pointers?